catalog/core/catalog-core-api/src/main/java/ddf/catalog/transform/MetacardEnricher.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transform;
+
+import java.io.InputStream;
+
+import ddf.catalog.data.Metacard;
+
+/**
+ * An MetacardEnricher extracts additional metadata from an {@code InputStream} and adds the
+ * metadata to a {@link Metacard}.
+ * <p>
+ * <b>Implementations of this interface <em>must</em>:</b>
+ * <ul>
+ * <li/>Register with the OSGi Service Registry using the {@link MetacardEnricher} interface
+ * <li/>Include a Service property with name "id" ({@link ddf.catalog.Constants#SERVICE_ID}) and a
+ * {@code String} value uniquely identifying the particular implementation
+ * </ul>
+ */
+public interface MetacardEnricher {
```

Comment:
I don't think that we actually need this interface. If we use the CatalogFramework.create(CreateStorageRequest) method, that will allow you to ingest a microsoft document, generate a metacard from it using your transformer and then as the document itself is getting added to the content store, you can parse it in a storage plugin to add additional fields to the metacard. The PostCreateStoragePlugins hand you a list of content items and each content item has a metacard and an InputStream, which is exactly what this interface uses.